x-pack/plugin/core/template-resources/src/main/resources/ml/config_index_mappings.json

```diff
@@ -376,6 +376,9 @@
           }
         }
       },
+      "esql_query": {
+        "type": "keyword"
+      },
```

Comment:
As well as this you need to bump this constant to 2: https://github.com/elastic/elasticsearch/blob/fe02cf0b2bb2ba9dcab89ee724c4a5481fa20ac4/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlConfigIndex.java#L23

Reply:
What is this exactly used for?

Reply:
That number gets embedded into the index mappings, and is then used to work out if they need updating. If this number is not incremented then the new mapping you've added won't get used in upgraded clusters. (It _will_ get used in brand new clusters, which might mask the problem.)

Reply:
Thanks!